index.d.ts

```diff
@@ -0,0 +1,170 @@
+declare module 'fluture' {
+  interface ResolveFunction<R> {
+    (value: R): void
+  }
+  interface RejectFunction<L> {
+    (error: L): void
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork(reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R>(duration: number, value: R): IFuture<L, R>
+
+  export function and<L, R>(
+    left: IFuture<L, any>,
+    right: IFuture<L, R>
+  ): IFuture<L, R>
+
+  export function ap<L, R, VR>(
+    apply: IFuture<L, (value: VR) => R>,
+    value: IFuture<L, VR>
+  ): IFuture<L, R>
+
+  export function attempt<L, R>(f: () => L): IFuture<L, R>
+
+  export function bimap<A, B, AA, BB>(
+    left: (a: A) => AA,
+    right: (a: B) => BB
+  ): IFuture<AA, any>
```

Comment:
Let's use `LA`, `LB`, `RA`, and `RB`. I think it's also nice to use names that correspond to the variable names used in Fluture's source-code. Furthermore, this function is missing its third parameter.

```ts
bimap<LA, LB, RA, RB>(
  lmapper: (reason: LA) => LB),
  rmapper: (value: RA) => RB),
  source: IFuture<LA, RA>
): IFuture<LB, RB>
```